source/as-promise/index.ts

```diff
@@ -22,13 +23,30 @@ const proxiedRequestEvents = [
 	'downloadProgress',
 ];
 
-export default function asPromise<T>(firstRequest?: Request): CancelableRequest<T> {
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const getDOMException = (errorMessage: string) => globalThis.DOMException === undefined
+	? new AbortError(errorMessage)
+	: new DOMException(errorMessage);
+
+export default function asPromise<T>(firstRequest?: Request, signal?: AbortSignal): CancelableRequest<T> {
 	let globalRequest: Request;
 	let globalResponse: Response;
 	let normalizedOptions: Options;
 	const emitter = new EventEmitter();
 
 	const promise = new PCancelable<T>((resolve, reject, onCancel) => {
+		if (signal) {
```

Comment:
Add a TODO comment to use `signal.throwIfAborted()` when targeting Node.js 18. That would simplify the code a lot. https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted

Same with your other pull requests.

Reply:
I think throwing error here prevents error catching using `Promise#catch`.

Shouldn't we call `reject(signal.reason);` instead? (when targeting Node v18)


> Add a TODO comment to use `signal.throwIfAborted()` when targeting Node.js 18. That would simplify the code a lot. [developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted)
> 
> Same with your other pull requests.

I think throwing error here prevents error catching using `Promise#catch`.

Shouldn't we call `reject(signal.reason);` instead? (when targeting Node v18)

Reply:
Actually both, if you look at the example on MDN: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted#implementing_an_abortable_api

Reply:
I think the example code was incorrect.
You can see about this on https://github.com/mdn/content/issues/15348 
So we won't need to call `throwIfAborted` here.